network/hive.go

```diff
@@ -390,24 +390,30 @@ func (h *Hive) handlePeersMsg(d *Peer, msg *peersMsg) error {
 func (h *Hive) handleSubPeersMsg(ctx context.Context, d *Peer, msg *subPeersMsg) error {
 	d.setDepth(msg.Depth)
 	// only send peers after the initial subPeersMsg
-	if !d.sentPeers {
-		var peers []*BzzAddr
-		// iterate connection in ascending order of disctance from the remote address
-		h.EachConn(d.Over(), 255, func(p *Peer, po int) bool {
-			// terminate if we are beyond the radius
-			if uint8(po) < msg.Depth {
-				return false
-			}
-			if !d.seen(p.BzzAddr) { // here just records the peer sent
-				peers = append(peers, p.BzzAddr)
-			}
-			return true
-		})
-		// if useful  peers are found, send them over
-		if len(peers) > 0 {
-			go d.Send(ctx, &peersMsg{Peers: sortPeers(peers)})
-		}
+	h.lock.Lock()
+	if d.sentPeers {
```

Comment:
@pradovic please run all `hive` tests multiple times (with high `count` values). i've made some changes to hive message handlers at the time and get tests to start flaking for some very bizarre reasons (which to date i have not managed to understand why)

Reply:
Sure, I will, thanks! I have run them multiple times as is, but I will try to increase count values more and see what happens.

Reply:
Hey @acud I could not get hive tests to fail. Do you have any particular test and counter  combination in mind? The reason for the lock in this place is that this code is now async.